lib/WiiExtension/extensions/ClassicExtension.cpp

```diff
@@ -0,0 +1,210 @@
+        buttons[WiiButtons::WII_BUTTON_R]                      = !((inputData[7] & 0x02) >> 1);
+
+        buttons[WiiButtons::WII_BUTTON_ZL]                     = !((inputData[8] & 0x80) >> 7);
+        buttons[WiiButtons::WII_BUTTON_B]                      = !((inputData[8] & 0x40) >> 6);
+        buttons[WiiButtons::WII_BUTTON_Y]                      = !((inputData[8] & 0x20) >> 5);
+        buttons[WiiButtons::WII_BUTTON_A]                      = !((inputData[8] & 0x10) >> 4);
+        buttons[WiiButtons::WII_BUTTON_X]                      = !((inputData[8] & 0x08) >> 3);
+        buttons[WiiButtons::WII_BUTTON_ZR]                     = !((inputData[8] & 0x04) >> 2);
+        buttons[WiiButtons::WII_BUTTON_LEFT]                   = !((inputData[8] & 0x02) >> 1);
+        buttons[WiiButtons::WII_BUTTON_UP]                     = !((inputData[8] & 0x01) >> 0);
+    } else if (getDataType() == WII_DATA_TYPE_3) {
+        analogState[WiiAnalogs::WII_ANALOG_LEFT_X]             = (inputData[0] & 0xFF);
+        analogState[WiiAnalogs::WII_ANALOG_LEFT_Y]             = (inputData[2] & 0xFF);
+        analogState[WiiAnalogs::WII_ANALOG_RIGHT_X]            = (inputData[1] & 0xFF);
+        analogState[WiiAnalogs::WII_ANALOG_RIGHT_Y]            = (inputData[3] & 0xFF);
+
+        analogState[WiiAnalogs::WII_ANALOG_LEFT_TRIGGER]       = (inputData[4] & 0xFF);
+        analogState[WiiAnalogs::WII_ANALOG_RIGHT_TRIGGER]      = (inputData[5] & 0xFF);
+
+        buttons[WiiButtons::WII_BUTTON_RIGHT]                  = !((inputData[6] & 0x80) >> 7);
+        buttons[WiiButtons::WII_BUTTON_DOWN]                   = !((inputData[6] & 0x40) >> 6);
+        buttons[WiiButtons::WII_BUTTON_L]                      = !((inputData[6] & 0x20) >> 5);
+        buttons[WiiButtons::WII_BUTTON_MINUS]                  = !((inputData[6] & 0x10) >> 4);
+        buttons[WiiButtons::WII_BUTTON_HOME]                   = !((inputData[6] & 0x08) >> 3);
+        buttons[WiiButtons::WII_BUTTON_PLUS]                   = !((inputData[6] & 0x04) >> 2);
+        buttons[WiiButtons::WII_BUTTON_R]                      = !((inputData[6] & 0x02) >> 1);
+
+        buttons[WiiButtons::WII_BUTTON_ZL]                     = !((inputData[7] & 0x80) >> 7);
+        buttons[WiiButtons::WII_BUTTON_B]                      = !((inputData[7] & 0x40) >> 6);
+        buttons[WiiButtons::WII_BUTTON_Y]                      = !((inputData[7] & 0x20) >> 5);
+        buttons[WiiButtons::WII_BUTTON_A]                      = !((inputData[7] & 0x10) >> 4);
+        buttons[WiiButtons::WII_BUTTON_X]                      = !((inputData[7] & 0x08) >> 3);
+        buttons[WiiButtons::WII_BUTTON_ZR]                     = !((inputData[7] & 0x04) >> 2);
+        buttons[WiiButtons::WII_BUTTON_LEFT]                   = !((inputData[7] & 0x02) >> 1);
+        buttons[WiiButtons::WII_BUTTON_UP]                     = !((inputData[7] & 0x01) >> 0);
+    } else {
+        // unknown
+    }
+
+#if WII_EXTENSION_DEBUG==true
```

Comment:
Just for cleanliness, we'll probably want to remove this before release if you're not using it. Okay to keep it into this refactor!